### PR TITLE
feature: add argument connections in process enable_privileged_agent

### PR DESCRIPTION
**Pull request by yaolaiyaoqu**
add argument connections in process enable_privileged_agent

### CODE REVIEW

**Comment by spacewander** [comment]
We need test for new feature.

File: lib/ngx/process.lua [hunk 1]
```diff
@@ -43,12 +43,20 @@ function _M.type()
...
         return nil, "API disabled in the current context"
     end
 
-    local rc = C.ngx_http_lua_ffi_enable_privileged_agent(errmsg)
+    if connections == nil then
+        connections = 512
+    end
+
+    if connections < 0 then
+        return nil, "Invaild connections"
```
Comment (doujiang24) [inline lib/ngx/process.lua:56]:
```suggestion
        return nil, "invalid connections"
```

File: t/process-type-privileged-agent-connections.t [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+type: worker
+--- error_log
+connect ok
+--- no_error_log
+connect not ok
+enable_privileged_agent failed
+--- skip_nginx: 5: < 1.11.2
+--- wait: 0.2
+
+
```
Comment (doujiang24) [inline t/process-type-privileged-agent-connections.t:78]: need three blank lines here, and you are recommended to use `reindex` tool to reformat it automatically.
https://github.com/openresty/openresty-devel-utils#reindex

File: t/process-type-privileged-agent-connections.t [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+        end
+    }
+
+    init_worker_by_lua_block {
+        local base = require "resty.core.base"
+        local v
+        local typ = (require "ngx.process").type
+        for i = 1, 400 do
+            v = typ()
+        end
```
Comment (xiaobiaozhao) [inline t/process-type-privileged-agent-connections.t:37]:
```
       for i = 1, 400 do
            v = typ()
        end
```
What does this do?

**Review by xiaobiaozhao — COMMENTED** [review]
What problem do we need to introduce this PR?

**Comment by yaolaiyaoqu** [comment]
the pr is base on other pr
https://github.com/openresty/stream-lua-nginx-module/pull/255
https://github.com/openresty/lua-nginx-module/pull/1774
https://github.com/openresty/openresty/pull/686

File: t/process-type-privileged-agent-connections.t [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    init_worker_by_lua_block {
+        local base = require "resty.core.base"
+        local typ = (require "ngx.process").type()
+
+        if typ == "privileged agent" then
+            ngx.log(ngx.WARN, "process type: ", typ)
+            ngx.timer.at(0, function()
+
+                for i = 1, 10 do
+                    local tcpsock = ngx.socket.tcp()
```
Comment (zhuizhuhaomeng) [inline t/process-type-privileged-agent-connections.t:92]: reach limit, but not exceed. why failed?

File: t/process-type-privileged-agent.t [hunk 5]
```diff
@@ -43,6 +43,21 @@ add_block_preprocessor(sub {
 
         if v == "privileged agent" then
             ngx.log(ngx.WARN, "process type: ", v)
+
+            ngx.timer.at(0, function() 
+   
+                for i = 1, 4 do 
+                    local tcpsock = ngx.socket.tcp()
```
Comment (zhuizhuhaomeng) [inline t/process-type-privileged-agent.t:50]: aliginment

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: t/process-type-privileged-agent.t (hunk 5 above)
Comment (yaolaiyaoqu) [reply t/process-type-privileged-agent.t:50]: for is the body of a function argument of ngx.timer.at， then need ident

File: t/process-type-privileged-agent-connections.t (hunk 4 above)
Comment (yaolaiyaoqu) [reply t/process-type-privileged-agent-connections.t:92]: i found openresty will create a little socket when you start. but i did'nt deep research.

File: lib/ngx/process.lua [hunk 6]
```diff
@@ -73,12 +73,18 @@ function _M.type()
 end
 
 
-function _M.enable_privileged_agent()
+function _M.enable_privileged_agent(connections)
     if ngx_phase() ~= "init" then
         return nil, "API disabled in the current context"
```
[the remark this answers is not in the document]
Comment (yaolaiyaoqu) [reply lib/ngx/process.lua:80]: ok, is done.

**Review by doujiang24 — APPROVED** [review]

**Pull request merged**